### PR TITLE
fix: typescript errors

### PR DESCRIPTION
**Pull request by pecoram**

### CODE REVIEW

**Comment by chiefcll** [comment]
This looks good. @anthony9187 - want to check it out as well? I'll check these changes in the demo-app

**Review by chiefcll — APPROVED** [review]

File: src/core/node/elementNode.ts [hunk 1]
```diff
@@ -457,12 +460,8 @@ export class ElementNode extends Object {
...
       let changedLayout = false;
       if (isFunc(this.onBeforeLayout)) {
-        changedLayout = this.onBeforeLayout.call(
-          this,
-          this,
-          child,
-          dimensions,
-        ) as boolean;
+        changedLayout =
+          this.onBeforeLayout.call(this, this, child, dimensions) ?? false;
```
Comment (anthony9187) [inline src/core/node/elementNode.ts]: im not sure if this is a my app problem, but when i try to build this into solid-ui i get the following error here:

```
../../../../solid/src/core/node/elementNode.ts:463:9 - error TS2322: Type 'boolean | void' is not assignable to type 'boolean'.
  Type 'void' is not assignable to type 'boolean'.
  ```
  
  it seems like `??` should guard against this but the compiler still doesn't like it

Comment (pecoram) [reply src/core/node/elementNode.ts]: I've just tried to build solid-ui with local solid folder and don't see this error

Comment (pecoram) [reply src/core/node/elementNode.ts]: I got it running tsc.. maybe some tsconfig configuration

Comment (pecoram) [reply src/core/node/elementNode.ts]: I relplaced ?? with ||

Comment (anthony9187) [reply src/core/node/elementNode.ts]: ugh that's my bad, i should've been clear, i was testing in a branch where I'm converting to tsc 😅 thanks! cleared it up for me as well

**Review by anthony9187 — APPROVED** [review]

**Pull request merged**